powertools-sqs/src/main/java/software/amazon/lambda/powertools/sqs/PowertoolsSqs.java

```diff
@@ -74,12 +80,192 @@ private PowertoolsSqs() {
+     * @return List of values returned by {@link SqsMessageHandler#process(SQSMessage)} while processing each message-
+     * @throws SQSBatchProcessingException if some messages fail during processing.
+     */
+    public static <R> List<R> batchProcessor(final SQSEvent event,
+                                             final SqsMessageHandler<R> handler) {
+        return batchProcessor(event, false, handler);
+    }
+
+    /**
+     * This utility method is used to processes each {@link SQSMessage} inside received {@link SQSEvent}
+     *
+     * <p>
+     * Utility will take care of calling {@link SqsMessageHandler#process(SQSMessage)} method for each {@link SQSMessage}
+     * in the received {@link SQSEvent}
+     * </p>
+     *
+     * </p>
+     * If any exception is thrown from {@link SqsMessageHandler#process(SQSMessage)} during processing of a messages,
+     * Utility will take care of deleting all the successful messages from SQS. When one or more single message fails
+     * processing due to exception thrown from {@link SqsMessageHandler#process(SQSMessage)}
+     * {@link SQSBatchProcessingException} is thrown with all the details of successful and failed messages.
+     * <p>
+     * Exception can also be suppressed if desired.
+     * <p>
+     * If all the messages are successfully processes, No SQS messages are deleted explicitly but is rather delegated to
+     * Lambda execution context for deletion.
+     * </p>
+     *
+     * @param event             {@link SQSEvent} received by lambda function.
+     * @param suppressException if this is set to true, No {@link SQSBatchProcessingException} is thrown even on failed
+     *                          messages.
+     * @param handler           Instance of class implementing {@link SqsMessageHandler} which will be called for each message in event.
+     * @return List of values returned by {@link SqsMessageHandler#process(SQSMessage)} while processing each message.
+     * @throws SQSBatchProcessingException if some messages fail during processing and no suppression enabled.
+     */
+    public static <R> List<R> batchProcessor(final SQSEvent event,
+                                             final boolean suppressException,
+                                             final SqsMessageHandler<R> handler) {
+        final List<R> handlerReturn = new ArrayList<>();
+
+        BatchContext batchContext = new BatchContext(defaultSqsClient());
+
+        for (SQSMessage message : event.getRecords()) {
+            try {
+                handlerReturn.add(handler.process(message));
+                batchContext.addSuccess(message);
+            } catch (Exception e) {
+                batchContext.addFailure(message, e);
+            }
+        }
+
+        batchContext.processSuccessAndHandleFailed(handlerReturn, suppressException);
+
+        return handlerReturn;
+    }
+
+    private static SqsClient defaultSqsClient() {
```

Comment:
change to `sqsClient()`?

Reply:
I removed it altogether an just used the class variable since it was used in just one place anyways